Sources/ProtobufferCoding/Decoder/UnkeyedProtoDecodingContainer.swift

```diff
@@ -81,7 +81,14 @@ class UnkeyedProtoDecodingContainer: InternalProtoDecodingContainer, UnkeyedDeco
     }
 
     func decode(_ type: Int.Type) throws -> Int {
-        throw ProtoError.decodingError("Int not supported, use Int32 or Int64")
+        if let value = popNext().last {
+            if MemoryLayout<Int>.size == 4 {
+                return try Int(decodeInt32(value))
+            } else if MemoryLayout<Int>.size == 8 {
+                return try Int(decodeInt64(value))
+            }
```

Comment:
Similar to the comment above I think that this file could also benefit from an extension of the `Int` type. The sea applies to other `Int`ish types 👍